### PR TITLE
RemoteCache: Support SSL with Redis

### PR DESCRIPTION
**Pull request by kylebrandt**
For https://github.com/grafana/grafana/issues/18498 

- [x] update redis lib from v2 -> v5 (v6 seems to have issue with go modules that will be fixed in 7 (commit 4 days ago).
- [x] make sure updated lib works as is with redis
- [x] Add SSL param to connection string and tests for it
- [x] Add TLSConfig to client when there is a redis client
- [x] Manually make sure it works with Azure Redis 
- [x] Update Documentation
- [x] Update ini files

### CODE REVIEW

**Comment by kylebrandt** [comment]
Errr... I did v5 not v6

File: docs/sources/installation/configuration.md [hunk 1]
```diff
@@ -281,9 +281,24 @@ Either `redis`, `memcached` or `database` default is `database`
 
 ### connstr
 
-The remote cache connection string. Leave empty when using `database` since it will use the primary database.
-Redis example config: `addr=127.0.0.1:6379,pool_size=100,db=grafana`
-Memcache example: `127.0.0.1:11211`
+The remote cache connection string. The format depends on the `type` of the remote cache.
```
Comment (kylebrandt) [inline docs/sources/installation/configuration.md]: Note: have not previewed these changes yet - issues running docs locally.

File: pkg/infra/remotecache/redis_storage.go [hunk 2]
```diff
@@ -71,7 +82,7 @@ func (s *redisStorage) Set(key string, val interface{}, expires time.Duration) e
 	if err != nil {
 		return err
 	}
-	status := s.c.SetEx(key, expires, string(value))
+	status := s.c.Set(key, string(value), expires)
```
Comment (kylebrandt) [inline pkg/infra/remotecache/redis_storage.go]: SetEx is gone from the newer version of the redis lib and replaced with Set with always takes an expire param.

File: pkg/infra/remotecache/redis_storage.go [hunk 3]
```diff
@@ -38,7 +40,7 @@ func parseRedisConnStr(connStr string) (*redis.Options, error) {
 		case "password":
 			options.Password = connVal
 		case "db":
-			i, err := strconv.ParseInt(connVal, 10, 64)
+			i, err := strconv.Atoi(connVal)
```
Comment (kylebrandt) [inline pkg/infra/remotecache/redis_storage.go]: note: redis lib changed type from int64 to int

File: pkg/infra/remotecache/redis_storage.go [hunk 4]
```diff
@@ -49,9 +51,27 @@ func parseRedisConnStr(connStr string) (*redis.Options, error) {
...
 		default:
-			return nil, fmt.Errorf("unrecorgnized option '%v' in redis connection string", connVal)
+			return nil, fmt.Errorf("unrecognized option '%v' in redis connection string", connKey)
+		}
+	}
+	if setTLSIsTrue {
+		// Get hostname from the Addr property and set it on the configuration for TLS
+		sp := strings.Split(options.Addr, ":")
+		if len(sp) < 1 {
+			return nil, fmt.Errorf("unable able to get hostname from the addr field, expected host:port, got '%v'", options.Addr)
```
Comment (marefr) [inline pkg/infra/remotecache/redis_storage.go]:
```suggestion
			return nil, fmt.Errorf("unable to get hostname from the addr field, expected host:port, got '%v'", options.Addr)
```

**Review by marefr — APPROVED** [review]
LGTM besides a minor change suggestion

**Pull request merged**